accepted/2021/InboxSourceGenerators.md

```diff
@@ -0,0 +1,187 @@
+
+Generated code should not be generated in System namespaces, it should prefer the user's namespace.  Generated code should be triple-slashed commented and be free from compiler warnings.  Generated code should avoid name collisions with user code and provide affordances for resolving name conflicts.  As much as possible generated code should meet the quality standards of .NET library code, using nullable annotations, meeting framework design guidelines, naming guidelines, etc.  It should feel like a natural extension of the library.
+
+### Deterministic failure
+
+The absence of a required source generator from code that requires it should produce a clear diagnosable error message from the compiler, like the error from a missing reference.  A codebase that was written to make use of a source-generator should never compile and silently omit the source generated code.  For example: if a source generator uses and attribute to activate the generator and then generates extension methods with overloads that are automatically preferred, the source generator should define the attribute in the source generator, not the framework library.  This will ensure that the project fails to compile if the source generator is absent.
+
+### Source generator servicing
+
+Source generators contribute significant code to user assemblies and that code may have issues.  We have very little control over changing customer assemblies once they have consumed a source generator.  We cannot and should not attempt to change the behavior of customer's assemblies (eg: hot patch) to fix issues.  Should an issue be found we should fix it in the source generator and ship a new version, just as we would if we found an issue with compiler.  Customers desiring this fix must proactively acquire the new code-generator, typically through an updated SDK which carries the ref pack, rebuild their code, and update their production applications.
+
+Due to the limited servicing agility for code-generated code, we should be very careful about the complexity of the code which is generated.  Where possible we should limit generated code to "glue code", only that which must be specific to the user's assembly or types.  We should try to put more complex code inside the framework or library itself, even if it means exposing public API that is specifically for source generated code to call.
+
+Inbox source generators will be serviced in their side-by-side location.  Should a bug exist in a source generator we will evaluate the bug against the bar for each framework it exists and patch them independently.  In this way the servicing process for these source generators will match the .NET runtime.
```

Comment:
The analyzer will be in it's own location.  Side-by-side in this context is talking about versions of the analyzer specific to each framework, rather than a central version used for all frameworks.  I can avoid the term `side-by-side` here if it creates confusion.  It's a bit of a relic of how we talked about .NETFramework components as side-by-side vs shared.

Reply:
Scrubbed the document for uses of side by side and clarified them.